Minimum Ansible version check

mtnbikenc opened the pull request:
Resolves: https://github.com/openshift/openshift-ansible/issues/2925

Moved version check to callback pluging to allow checking the version
prior to playbook syntax checking. This prevents errors on
syntax-checking with older versions of Ansible.

mtnbikenc commented:
@detiber Here's a first shot at a new version check. I figure we would also remove the `verify_ansible_version.yml` playbook.

tbielawa commented on callback_plugins/ansible_version_check.py:
```diff
@@ -0,0 +1,33 @@
+    CallbackBase = object
+
+
+def print_red_bold(text):
```
There's a built-in with the callback system for printing 'fail' type text, using 'print' directly should be avoided for this. I'll come back with a proper code example in a few minutes.

tbielawa replied:
Set the message string and then use the built-in `_display` (it's a part of callback objects). Here's an example where they detect a task failed and then use the `C` constants to print with the correct red color:

https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/default.py#L55-L57

You'll have to make sure you initialize the parent class though: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/default.py#L41

to make sure that the `_display` gets set: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/__init__.py#L55

Or add the kwarg yourself to the ``__init__`` definition (I think).

mtnbikenc replied:
In looking through trying to implement this I've found that the more we rely on Ansible, while trying to check the current Ansible version, we would break on older versions of Ansible.  In this instance `C.COLOR_ERROR` does not exist in Ansible 2.0.0.2 and bombs. :(  So, it seems we should very lightly rely on Ansible dependencies.  Currently I'm using `__version__` which has been around since the beginning.  Thoughts?

tbielawa replied:
from `constants.py` it defines `COLOR_ERROR` with a default value of the string `red`. I think you could get away with calling `_display.display(...., color='red')`. The difference being that we are overriding any custom user configs (which we were doing anyway using the shell escape codes :) )

mtnbikenc commented:
@tbielawa I've updated the code and this works only on post 2.0.0.0 versions of Ansible, which is not what we want.  On ansible-1.9.6 I get:

```
AttributeError: 'CallbackModule' object has no attribute '_display'
```

mtnbikenc commented:
I've updated the code and tested back to Ansible v1.7 successfully.  Additionally, I have removed the verify_ansible_version playbook and references to the same.

tbielawa commented:
aos-ci-test

sdodson commented:
aos-ci-test

detiber commented:
@mtnbikenc one issue that I see is that there is no license on the code from the blog, and the blog lists 'All rights reserved' with the copywrite notice. I've emailed the author to see if he is willing to release a copy of the code with a permissive license.

mtnbikenc commented:
aos-ci-test

mtnbikenc commented:
aos-ci-test
